### PR TITLE
TST: Fix setitem parametrizations

### LINKED ISSUE #56727: TST: test_setitem_integer_with_missing_raises contains duplicate parametrization
https://github.com/pandas-dev/pandas/blob/c4b6bed5346521617b8361801cad0bf854c40994/pandas/tests/extension/base/setitem.py#L211-L216

One of these parametrizations should probably be `(pd.array([0, 1, 2, pd.NA], dtype="Int64"), True),` but that fails several tests that inherit this test. The task is to investigate whether the failure is expected or OK to xfail

### PR DESCRIPTION
**Pull request by lopof**
I'm not sure if i am on the right track with this PR but it's a try. If not i will close it and probably someone more into the code has to dig into that issue.

- [x] closes #56727
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -2051,6 +2054,8 @@ def setitem(self, indexer, value, using_cow: bool = False):
             # TODO(GH#45419): string[pyarrow] tests break if we transpose
             #  unconditionally
             values = values.T
+
+        check_array_indexer(values, indexer)
```
Comment (mroeschke) [inline pandas/core/internals/blocks.py:2058]: cc @jbrockmendel

**Comment by jbrockmendel** [comment]
I'd like to understand how this relates to #31948 (for which the behavior seems to have changed since it was opened).

Ideally we'd do checks on the indexer once at the DataFrame/Series level instead of at the block level which can end up doing them repeatedly.

**Comment by lopof** [comment]
@jbrockmendel i updated #31948 with the current behavior. What would be the desired behavior? Maybe the test has wrong expectations...

**Comment by lopof** [comment]
With `@pytest.mark.filterwarnings("ignore::FutureWarning")` instead of
```
if type(arr) is pd.Series:
    arr.iloc[idx] = arr.iloc[0]
else:
    arr[idx] = arr[0]
```
the main reason the test are failing is `KeyError: '[0 1 2 <NA>] not in index'` is raised. 
But this could also be the desired Error?

**Comment by lopof** [comment]
o opened a new PR, that only changes the `setitem-py`, but does does pass all tests. See #57629

**Comment by lopof** [comment]
@jbrockmendel @mroeschke maybe you can have a look at  #57629

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**